rzup/src/lib.rs

```diff
@@ -11,64 +11,378 @@
+    ///
+    /// # Arguments
+    /// * `component_id` - Component identifier
+    /// * `version` - Version to create directories for
+    pub fn ensure_version_dirs(&self, component_id: &str, version: &Version) -> Result<()> {
+        Paths::create_version_dirs(&self.environment, component_id, version)
+    }
+
+    /// Gets the component directory path.
+    ///
+    /// # Arguments
+    /// * `component_id` - Component identifier
+    pub fn get_component_dir(&self, component_id: &str) -> Result<PathBuf> {
+        Paths::get_component_dir(&self.environment, component_id)
+    }
+
+    /// Gets the version-specific directory path for a component.
+    ///
+    /// # Arguments
+    /// * `component_id` - Component identifier
+    /// * `version` - Version to get directory for
+    pub fn get_version_dir(&self, component_id: &str, version: &Version) -> Result<PathBuf> {
+        Paths::get_version_dir(&self.environment, component_id, version)
+    }
+}
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempfile::TempDir;
+
+    fn setup_test_env() -> (TempDir, Rzup) {
+        let tmp_dir = TempDir::new().unwrap();
+        let rzup = Rzup::with_root(tmp_dir.path()).unwrap();
+        (tmp_dir, rzup)
+    }
+
+    #[test]
+    fn test_rzup_initialization() {
+        let (_tmp_dir, rzup) = setup_test_env();
+        assert!(rzup.settings().get_active_version("rust").is_none());
+        assert!(rzup
+            .settings()
+            .get_active_version("cargo-risczero")
+            .is_none());
+    }
+
+    #[test]
+    fn test_path_operations() {
+        let (_tmp_dir, rzup) = setup_test_env();
+        let version = Version::new(1, 2, 0);
+        let component_id = "cargo-risczero";
+
+        // Create necessary directories without actual installation
+        rzup.ensure_version_dirs(component_id, &version).unwrap();
+
+        // Create the bin directory for the component
+        let bin_dir = rzup.get_version_dir(component_id, &version).unwrap();
+        std::fs::create_dir_all(&bin_dir).unwrap();
+
+        // Test binary path retrieval with platform-specific path
+        let bin_path = rzup.get_bin_path(component_id, &version).unwrap();
+        assert!(bin_path.is_some());
+        let bin_path = bin_path.unwrap();
+        assert!(bin_path.ends_with(format!(
+            "v{}-{}-{}/bin",
+            version,
+            component_id,
+            rzup.environment.platform()
+        )));
+
+        // Test virtual component (r0vm is inside cargo-risczero's bin directory)
+        let virtual_component = "r0vm";
+        let virtual_bin_path = rzup.get_bin_path(virtual_component, &version).unwrap();
+        assert!(virtual_bin_path.is_some());
+        let virtual_bin_path = virtual_bin_path.unwrap();
+        assert!(virtual_bin_path.ends_with(format!("bin/{}", virtual_component)));
+    }
+
+    #[test]
+    #[ignore = "requires GitHub API access"]
```

Comment:
My initial reaction to tests like this are to wonder if there is a way we can run them without doing the remote network access piece.

In the past I've gotten around this in different ways,
one way is to allow passing in the base URLs that are being accessed and in your tests you run a little HTTP server and pass in that address, another way is just dependency injecting the part that does the HTTP requests. What are your thoughts on these sorts of approaces?